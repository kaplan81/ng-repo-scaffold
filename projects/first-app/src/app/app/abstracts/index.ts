export * from '@first-app/abstracts/component.abstract';
export * from '@first-app/abstracts/container.abstract';
export * from '@first-app/abstracts/guard.abstract';
