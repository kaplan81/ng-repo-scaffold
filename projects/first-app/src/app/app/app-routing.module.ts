import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import * as fromRootEnums from '@first-app/enums';

const redirectTo = `/${fromRootEnums.NavPrimaryPath.feature1}/${fromRootEnums.NavChildrenPath.entity1}`;

const routes: Routes = [
  {
    path: fromRootEnums.NavPrimaryPath.feature1,
    loadChildren: '../feature1/feature1.module#Feature1Module'
  },
  {
    path: fromRootEnums.NavPrimaryPath.wildcard,
    redirectTo
  }
];

@NgModule({
  // Uncomment the next to enable tracing and to see what events are happening during the navigation lifecycle.
  // imports: [RouterModule.forRoot(routes, { enableTracing: true })],
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}
