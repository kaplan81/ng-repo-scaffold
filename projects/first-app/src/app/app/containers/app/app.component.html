<div id="fstlayout">
  <header id="fst-header">
    <nav id="fst-nav"></nav>
  </header>
  <aside id="fst-sidebar"></aside>
  <main id="fst-main">
    <router-outlet></router-outlet>
  </main>
</div>
