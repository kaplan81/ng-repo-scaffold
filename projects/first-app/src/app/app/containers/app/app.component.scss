@import 'variables';
@import 'mixins';

/* Provional layout for container in page so in <router-outlet> */
#fstlayout {
  display: grid;
  grid-template-rows: $header-height auto;
  grid-template-areas:
    'header'
    'page';
  height: 100%;
  margin: 0 $gap * 2;
  #fstheader {
    grid-area: header;
    display: grid;
    grid-template-columns: $sidebar-width auto;
    grid-template-areas: '. nav';
    height: $header-height;
    #fstnav {
      grid-area: nav;
      display: flex;
      padding-left: $gutter;
      h2 {
        align-self: center;
        margin-right: $h2;
      }
    }
  }
  #fstpage {
    grid-area: page;
    margin-bottom: $gap * 2;
    #fstcontainer {
      display: grid;
      grid-template-columns: $sidebar-width auto;
      grid-template-rows: auto;
      grid-template-areas:
      'sidebar main';
      height: 100%;
      #fstsidebar {
        grid-area: sidebar;
        .widget {
          @include borderDiv();
          margin: 0 $gutter 0 0;
          padding: $gap;
          .widget__heading {
            display: flex;
            height: 50px;
            display: flex;
            h3 {
              align-self: center;
            }
          }
          .widget__tags {
            padding: $gap * 2 0;
          }
        }
      }
      #fstmain {
        grid-area: main;
      }
    }
  }
}

