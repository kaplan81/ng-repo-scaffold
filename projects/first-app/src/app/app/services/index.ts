export * from '@first-app/services/api-base.service';
export * from '@first-app/services/guard.service';
export * from '@first-app/services/nav.service';
export * from '@first-app/services/router-state.service';
