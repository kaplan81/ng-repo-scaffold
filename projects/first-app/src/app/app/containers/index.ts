import { AppComponent } from '@first-app/containers/app/app.component';

export const containers: any[] = [AppComponent];

export { AppComponent } from '@first-app/containers/app/app.component';
