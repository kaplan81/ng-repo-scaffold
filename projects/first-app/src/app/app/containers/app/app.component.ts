import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as fromRootModels from '@first-app/models';
import * as fromRootServices from '@first-app/services';

@Component({
  selector: 'fst-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AppComponent {
  links: fromRootModels.Link[];

  constructor(private navService: fromRootServices.NavService) {
    this.links = [this.navService.feature1.link];

    // Tracing navs here.
    // console.log('this.navService:::', this.navService);
  }
}
