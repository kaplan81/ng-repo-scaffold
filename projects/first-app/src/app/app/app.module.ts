import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { CoreModule } from '@first-app-core/core.module';
import { environment } from '@first-app-environments/environment';
import { AppRoutingModule } from '@first-app/app-routing.module';
// import * as fromRootComponents from '@first-app/components';
import * as fromRootContainers from '@first-app/containers';
import { CustomSerializer, effects, reducers } from '@first-app/store';
import { EffectsModule } from '@ngrx/effects';
import { RouterStateSerializer, StoreRouterConnectingModule } from '@ngrx/router-store';
import { MetaReducer, StoreModule } from '@ngrx/store';
// Not used in production
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { storeFreeze } from 'ngrx-store-freeze';

export const metaReducers: MetaReducer<any>[] = !environment.production ? [storeFreeze] : [];

@NgModule({
  // declarations: [...fromRootContainers.containers, ...fromComponents.components],
  declarations: [...fromRootContainers.containers],
  imports: [
    BrowserModule,
    HttpClientModule,
    CoreModule,
    AppRoutingModule,
    StoreModule.forRoot(reducers, { metaReducers }),
    EffectsModule.forRoot(effects),
    StoreRouterConnectingModule,
    !environment.production ? StoreDevtoolsModule.instrument() : []
  ],
  providers: [{ provide: RouterStateSerializer, useClass: CustomSerializer }],
  bootstrap: [fromRootContainers.AppComponent]
})
export class AppModule {}
