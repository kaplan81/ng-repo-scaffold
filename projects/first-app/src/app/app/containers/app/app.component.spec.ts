/* tslint:disable:no-unused-variable */
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '@first-app/containers/app/app.component';
import * as fromRootServices from '@first-app/services';
import { ComponentSuite, ComponentSuiteElements } from '@project-scope/test-kit';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let els: ComponentSuiteElements<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AppComponent],
      imports: [RouterTestingModule],
      providers: [fromRootServices.NavService]
    })
      .compileComponents()
      .then(() => {
        fixture = TestBed.createComponent(AppComponent);
        els = new ComponentSuite<AppComponent>(fixture).elements;
      });
  }));

  it('should create', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
