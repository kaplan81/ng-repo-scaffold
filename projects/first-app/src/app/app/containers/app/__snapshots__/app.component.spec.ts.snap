// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create 1`] = `
<fst-root
  links={[Function Array]}
  navService={[Function NavService]}
>
  <div
    id="fstlayout"
  >
    <div
      id="fstheader"
    >
      <nav
        id="fstnav"
      >
        
        <h2
          ng-reflect-router-link="/,feature1"
          tabindex="0"
        >
          feature1
        </h2>
      </nav>
    </div>
    <div
      id="fstpage"
    >
      <div
        id="fstcontainer"
      >
        <div
          id="fstsidebar"
        >
          <div
            class="widget"
          >
            <div
              class="widget__heading"
            >
              <h3>
                Widget
              </h3>
            </div>
            <div
              class="widget__tags"
            >
              <h4>
                Tags
              </h4>
              <ul
                style="margin-top: 20px;"
              >
                <li>
                  Tag 1
                </li>
                <li>
                  Tag 2
                </li>
                <li>
                  Tag 3
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          id="fstmain"
        >
          <router-outlet />
        </div>
      </div>
    </div>
  </div>
</fst-root>
`;
